cpp/include/cuvs/preprocessing/quantization.hpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_mdarray.hpp>
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/handle.hpp>
+#include <raft/core/host_mdarray.hpp>
+#include <raft/core/host_mdspan.hpp>
+
+#include <cuda_fp16.h>
+
+namespace cuvs::preprocessing::quantization {
+
+/**
+ * @brief ScalarQuantizer parameters.
+ */
+struct sq_params {
```

Comment:
Please namespace to cuvs::preprocessing::quantize::scalar. Then this struct just becomes "params" nested in the scalar namespace. We are going to have several algorithms for quantizing and this will make it more straightforward for users.

Also please put create a directory for "quantize" and call this scalar.hpp.